src/java.base/share/classes/java/util/concurrent/Joiners.java

```diff
@@ -0,0 +1,220 @@
+        private final List<Subtask<T>> subtasks = new ArrayList<>();
+
+        private volatile Throwable firstException;
+
+        @Override
+        public boolean onFork(Subtask<? extends T> subtask) {
+            ensureUnavailable(subtask);
+            @SuppressWarnings("unchecked")
+            var s = (Subtask<T>) subtask;
+            subtasks.add(s);
+            return false;
+        }
+
+        @Override
+        public boolean onComplete(Subtask<? extends T> subtask) {
+            Subtask.State state = ensureCompleted(subtask);
+            return (state == Subtask.State.FAILED)
+                    && (firstException == null)
+                    && FIRST_EXCEPTION.compareAndSet(this, null, subtask.exception());
+        }
+
+        @Override
+        public Stream<Subtask<T>> result() throws Throwable {
+            Throwable ex = firstException;
+            if (ex != null) {
+                throw ex;
+            } else {
+                return subtasks.stream();
+            }
+        }
+    }
+
+    /**
+     * A joiner that returns the result of the first subtask to complete successfully.
+     * Cancels the scope if any subtasks succeeds.
+     */
+    static final class AnySuccessful<T> implements Joiner<T, T> {
+        private static final VarHandle SUBTASK =
+                MhUtil.findVarHandle(MethodHandles.lookup(), "subtask", Subtask.class);
+
+        // UNAVAILABLE < FAILED < SUCCESS
+        private static final Comparator<Subtask.State> SUBTASK_STATE_COMPARATOR =
+                Comparator.comparingInt(AnySuccessful::stateToInt);
+
+        private volatile Subtask<T> subtask;
+
+        /**
+         * Maps a Subtask.State to an int that can be compared.
+         */
+        private static int stateToInt(Subtask.State s) {
+            return switch (s) {
+                case UNAVAILABLE -> 0;
+                case FAILED      -> 1;
+                case SUCCESS     -> 2;
+            };
+        }
+
+        @Override
+        public boolean onComplete(Subtask<? extends T> subtask) {
+            Subtask.State state = ensureCompleted(subtask);
+            Subtask<T> s;
+            while (((s = this.subtask) == null)
+                    || SUBTASK_STATE_COMPARATOR.compare(s.state(), state) < 0) {
+                if (SUBTASK.compareAndSet(this, s, subtask)) {
+                    return (state == Subtask.State.SUCCESS);
+                }
+            }
+            return false;
+        }
+
+        @Override
+        public T result() throws Throwable {
+            Subtask<T> subtask = this.subtask;
+            if (subtask == null) {
+                throw new NoSuchElementException("No subtasks completed");
+            }
+            return switch (subtask.state()) {
+                case SUCCESS -> subtask.get();
+                case FAILED -> throw subtask.exception();
+                default -> throw new InternalError();
+            };
```

Comment:
```suggestion
            return switch (subtask.state()) {
                case SUCCESS -> subtask.get();
                case FAILED  -> throw subtask.exception();
                default      -> throw new InternalError();
            };
```
The switch of the stateToInt method is aligned, so it should also be aligned here to keep the style consistent

Reply:
I'll look to re-align these in the loom repo for the next refresh.